File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.ts
function increasingTriplet(nums: number[]): boolean {
    if (nums.length < 3)
    {
        return false;
    }
    let min1 = Infinity;
    let min2 = Infinity;
    for (let i = 0;i<nums.length;i++){
        if (min1>=nums[i]){
            min1 = nums[i]
        }
        else if (min2>=nums[i]){
            min2 = nums[i]
        }
        else{
            return true;
        }
    }
    return false;
};

File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        min1,min2 = float("inf"),float("inf")
        for i in range(len(nums)):
            if min1>=nums[i]:
                min1 = nums[i]
            elif min2>=nums[i]:
                min2 = nums[i]
            else:
                return True
        return False